src/bridging/providers/across/util.ts

```diff
@@ -29,43 +37,117 @@ export function getTokenAddress(tokenSymbol: string, chainConfig: AcrossChainCon
   // const lpFee = suggestedFees.lpFee
 
+  // Apply slippage
+  const buyAmountAfterSlippage = applyBps(buyAmountAfterFee, slippageBps)
+
   return {
-    suggestedFees,
-    buyAmount: applyFee(amount, suggestedFees.totalRelayFee.pct),
-    feeBps: pctToBps(suggestedFees.totalRelayFee.pct),
-    slippageBps: 0, // TODO: Can I set slippage to zero? are quotes enforced?
+    beforeFee: {
+      sellAmount: sellAmountBeforeFee,
+      buyAmount: buyAmountBeforeFee, // Assuming the price is 1:1 (before fee). This is because we are exchanges the same asset
+    },
+    afterFee: {
+      sellAmount: sellAmountBeforeFee, // Sell amount does't change (fee is applied to the buy amount)
+      buyAmount: buyAmountAfterFee,
+    },
+    afterSlippage: {
+      sellAmount: sellAmountBeforeFee, // Sell amount does't change (slippage is applied to the buy amount)
+      buyAmount: buyAmountAfterSlippage,
+    },
+
+    costs: {
+      bridgingFee: {
+        feeBps: pctToBps(totalRelayerFeePct),
+        amountInSellCurrency: feeSellToken,
+        amountInBuyCurrency: feeBuyToken,
+      },
+    },
+    slippageBps,
   }
 }
 
 /**
- * pct is a string that represents a percentage.
+ * Assert that a percentage is valid.
+ */
+function assertValidPct(pct: bigint): void {
+  if (pct > PCT_100_PERCENT || pct < 0n) {
+    throw new Error('Fee cannot exceed 100% or be negative')
+  }
+}
+
+/**
+ * pct is a represents a percentage.
  *
  * Note: 1% is represented as 1e16, 100% is 1e18, 50% is 5e17, etc. These values are in the same format that the contract understands.
  *
  * Bps is a percentage in basis points (1/100th of a percent). For example, 1% is 100 bps.
  *
+ * @param pct - The percentage to convert to bps
+ * @returns The percentage in bps
+ * @throws If the percentage is greater than 100% or less than 0%
  */
-export function pctToBps(pct: string): number {
-  const pctBigInt = BigInt(pct)
-  return Number((pctBigInt * 10_000n) / PCT_100_PERCENT)
-}
+export function pctToBps(pct: bigint): number {
+  assertValidPct(pct)
 
-export function applyFee(amount: string, pct: string): string {
-  const amountBigInt = BigInt(amount)
-  const pctBigInt = BigInt(pct)
+  return Number((pct * 10_000n) / PCT_100_PERCENT)
+}
 
-  if (pctBigInt > PCT_100_PERCENT) {
-    throw new Error('Fee cannot exceed 100%')
-  }
+/**
+ * Apply a percentage fee to an amount.
+ *
+ * @param amount - The amount to apply the fee to
+ * @param pct - The percentage fee to apply
+ * @throws If the percentage fee is greater than 100% or less than 0%
+ * @returns The amount after the fee has been applied
+ */
+export function applyPctFee(amount: bigint, pct: bigint): bigint {
+  assertValidPct(pct)
 
   // Compute amount after fee: amount * (1 - pct / 1e18)
-  const amountAfterFee = (amountBigInt * (PCT_100_PERCENT - pctBigInt)) / PCT_100_PERCENT
+  const amountAfterFee = (amount * (PCT_100_PERCENT - pct)) / PCT_100_PERCENT
+
+  return amountAfterFee
```

Comment:
Nipick because I dislike single use vars:

```suggestion
  return (amount * (PCT_100_PERCENT - pct)) / PCT_100_PERCENT
```